keps/sig-api-machinery/4020-unknown-version-interoperability-proxy/README.md

```diff
@@ -180,9 +172,14 @@ described above, could result in GC seeing a 404 and assuming an object has been
 deleted; this could result in it deleting a subsequent object that it should
 not.
 
-This proposal will cause the GC to see the complete list of resources in
-discovery, and when it requests specific objects, see either the correct object
-or get a 503 (which it handles safely).
+This proposal will cause the GC to behave in the following way for the scenarios described:
+
+- Case 1: if the resource is being deleted in the newer version 
+    - Any actions taken on this resource during the upgrade will render useless after the upgrade when these objects are orphaned so we can act fast and proxy this request to the newer apiserver. Garbage collection will be triggered for a deleted resource, which is correct in this case
+- Case 2: if the resource is being introduced in the newer version
+    - If the resource is being introduced in the newer version - proxying the discovery request to the newer apiserver is anyway the right thing to do. Garbage collection will be blocked which is also correct
```

Comment:
Ah I just meant that GC would see that the objects of the newly introduced API are available (as the discovery doc returned by the newest apiserver would reflect so). And if such objects are being referenced, it would not attempt to delete those.